core/DataAccess/Model.php

```diff
@@ -95,39 +98,49 @@ public function getInvalidatedArchiveIdsSafeToDelete($archiveTable, array $idSit
     }
 
     /**
-     * @param $archiveTable
-     * @param $idSites
-     * @param $periodId
-     * @param $datesToDelete
+     * @param string $archiveTable Prefixed table name
+     * @param int[] $idSites
+     * @param string[][] $datesByPeriodType
+     * @param Segment $segment
+     * @return \Zend_Db_Statement
      * @throws Exception
      */
-    public function updateArchiveAsInvalidated($archiveTable, $idSites, $periodId, $datesToDelete)
+    public function updateArchiveAsInvalidated($archiveTable, $idSites, $datesByPeriodType, Segment $segment = null)
     {
-        $sql = $bind = array();
-        $datesToDelete = array_unique($datesToDelete);
-        foreach ($datesToDelete as $dateToDelete) {
-            $sql[] = '(date1 <= ? AND ? <= date2 AND name LIKE \'done%\')';
-            $bind[] = $dateToDelete;
-            $bind[] = $dateToDelete;
-        }
-        $sql = implode(" OR ", $sql);
+        $idSites = array_map('intval', $idSites);
```

Comment:
Maybe `Common::getSqlStringFieldsArray($idSites)` would be more explicit to prevent SQL injection? Works though too so we can definitely leave it like that. Maybe a private method that contains `$idSites = array_map('intval', $idSites); implode(", ", $idSites)` could be nice just to have a description/comment what it is doing and to make it clear it's about preventing SQL injection. Not needed though